.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,40 @@
+{
+  "name": "community.general devcontainer",
+  "image": "mcr.microsoft.com/devcontainers/python:3.14-bookworm",
+  "features": {
+    "ghcr.io/devcontainers/features/docker-in-docker:2": {}
+  },
+  "customizations": {
+    "vscode": {
+      "settings": {
+        "terminal.integrated.shell.linux": "/bin/bash",
+        "python.pythonPath": "/usr/local/bin/python",
+        "editor.defaultFormatter": "charliermarsh.ruff",
+        "editor.formatOnSave": true,
+        "files.autoSave": "afterDelay",
+        "files.eol": "\n",
+        "files.insertFinalNewline": true,
+        "files.trimFinalNewlines": true,
+        "files.trimTrailingWhitespace": true
+      },
+      "extensions": [
+        "charliermarsh.ruff",
+        "chrislajoie.vscode-modelines",
+        "eamodio.gitlens",
+        "esbenp.prettier-vscode",
+        "michelemelluso.gitignore",
```

Comment:
michelemelluso.gitignore doesn't sound like something we should force on the user either.